### PR TITLE
Update usaco-491.mdx

### PR DESCRIPTION
**Pull request by Sanjana-Reddy-Kakulavaram**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-491.mdx [hunk 1]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+
+#include <bits/stdc++.h>
+using namespace std;
+#define fastio ios_base::sync_with_stdio(false); cin.tie(NULL);
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:20]: don't think this is part of the allowed macros

File: solutions/usaco-491.mdx [hunk 2]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+
+unordered_map < int, vector<int> > mp;
+vector< vector<int> > dist;
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:23]:
```suggestion
unordered_map<int, vector<int>> mp;
vector<vector<int>> dist;

Comment (SansPapyrus683) [reply solutions/usaco-491.mdx:23]: similar for other generics you define

File: solutions/usaco-491.mdx [hunk 3]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+
+void bfs (int s, int ix)
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:25]:
```suggestion
void bfs(int s, int ix)
```

File: solutions/usaco-491.mdx [hunk 4]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+{
+    queue< pair<int, int> > q;
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:27]: tabs

File: solutions/usaco-491.mdx [hunk 5]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+            dist[ix][n] = p.second + 1;
+
+            q.push({n, dist[ix][n]});
+        }
+    }
+}
+
+int main()
+{
+    fastio;
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:48]: remove if this runs w/o

File: solutions/usaco-491.mdx [hunk 6]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+    int A = INT_MAX;
+
+    for (int i = 0; i < N; i++)
+    {
+        A = min(A, dist[0][i] * B + dist[1][i] * E + dist[2][i] * P);
+    }
+
+    cout << A << "\n";
+
+    return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:78]:
```suggestion
    cout << A << endl;
```

File: solutions/usaco-491.mdx [hunk 7]
```diff
@@ -10,6 +10,79 @@ author: Qi Wang
...
+        cin >> u >> v;
+        u -= 1;
+        v -= 1;
+        mp[u].push_back(v);
+        mp[v].push_back(u);
+    }
+
+    bfs(0, 0);
+    bfs(1, 1);
+    bfs(N-1, 2);
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:67]: what does this do? add comments

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-491.mdx (hunk 1 above)
Comment (Sanjana-Reddy-Kakulavaram) [reply solutions/usaco-491.mdx]: I have removed that from the code

File: solutions/usaco-491.mdx (hunk 2 above)
Comment (Sanjana-Reddy-Kakulavaram) [reply solutions/usaco-491.mdx]: I have made the required changes

**Comment by SansPapyrus683** [comment]
there's still some other unresolved comments

File: solutions/usaco-491.mdx (hunk 3 above)
Comment (Sanjana-Reddy-Kakulavaram) [reply solutions/usaco-491.mdx]: I have removed spaces

**Review by Sanjana-Reddy-Kakulavaram — COMMENTED** [review]
I have made all the required changes.

File: solutions/usaco-491.mdx (hunk 4 above)
Comment (Sanjana-Reddy-Kakulavaram) [reply solutions/usaco-491.mdx]: I have made the required changes

File: solutions/usaco-491.mdx [hunk 8]
```diff
@@ -23,53 +23,53 @@ vector<vector<int>> dist;
...
+	queue<pair<int, int>> q;
+	q.push({s, 0});
+	dist[ix][s] = 0;
+	while(!q.empty())
+	{
+		pair<int, int> p = q.front();
+		q.pop();
+		for(auto n : mp[p.first])
+		{
+			if(dist[ix][n] != -1){ continue; }
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:35]: uh braces on separate lines please
like
```suggestion
			if(dist[ix][n] != -1){
				continue;
			}
```

File: solutions/usaco-491.mdx [hunk 9]
```diff
@@ -23,53 +23,53 @@ vector<vector<int>> dist;
...
+		}
+	}
 }
 
 int main()
 {
-    int B, E, P, N, M; 
-    cin >> B >> E >> P >> N >> M;
+	int B, E, P, N, M; 
+    	cin >> B >> E >> P >> N >> M;
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:45]: there's still spaces in some places

File: solutions/usaco-491.mdx [hunk 10]
```diff
@@ -23,53 +23,53 @@ vector<vector<int>> dist;
...
 
-    dist.assign(4, vector<int>(N+1, -1));
+    	dist.assign(4, vector<int>(N+1, -1));
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:47]: spaces between operators

File: solutions/usaco-491.mdx [hunk 11]
```diff
@@ -23,53 +23,53 @@ vector<vector<int>> dist;
...
-    for(int i = 0; i < M; i++)
-    {
-        int u, v;
-        cin >> u >> v;
-        u -= 1;
-        v -= 1;
-        mp[u].push_back(v);
-        mp[v].push_back(u);
-    }
+    	for(int i = 0; i < M; i++)
```
Comment (SansPapyrus683) [inline solutions/usaco-491.mdx:49]:
```suggestion
    	for (int i = 0; i < M; i++)
```
do similar things for other control flow statements

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by Sanjana-Reddy-Kakulavaram** [comment]
Can I please know, when will my pull request gets merged

**Comment by SansPapyrus683** [comment]
@envyaims @ryanchou-dev can yall review

File: solutions/usaco-491.mdx [hunk 12]
```diff
@@ -10,6 +10,71 @@ author: Qi Wang
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+unordered_map<int, vector<int>> mp;
+vector<vector<int>> dist;
+
+void bfs(int s, int ix)
```
Comment (envyaims) [inline solutions/usaco-491.mdx:23]: what does s and ix stand for? write it out please!

File: solutions/usaco-491.mdx [hunk 13]
```diff
@@ -10,6 +10,71 @@ author: Qi Wang
...
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+unordered_map<int, vector<int>> mp;
```
Comment (envyaims) [inline solutions/usaco-491.mdx:20]: bad practice, why not just an array of vectors?
also `mp` doesnt really make sense, maybe name it `adj` or `graph`?

File: solutions/usaco-491.mdx [hunk 14]
```diff
@@ -10,6 +10,71 @@ author: Qi Wang
...
+		}
+	}
+}
+
+int main()
+{
+	int B, E, P, N, M; 
+    	cin >> B >> E >> P >> N >> M;
+
+    	dist.assign(4, vector<int>(N + 1, -1));
```
Comment (envyaims) [inline solutions/usaco-491.mdx:49]: why does it have size 4 when youre only using three?

File: solutions/usaco-491.mdx [hunk 15]
```diff
@@ -10,6 +10,71 @@ author: Qi Wang
...
+        	v -= 1;
+        	mp[u].push_back(v);
+        	mp[v].push_back(u);
+    	}
+
+    	bfs(0, 0);  // finding the distance of every node from node-0
+    	bfs(1, 1);  // finding the distance of every node from node-1
+    	bfs(N - 1, 2);  // finding the distance of every node from node-(N-1)
+
+    	int A = INT_MAX;
```
Comment (envyaims) [inline solutions/usaco-491.mdx:64]: what does `A` stand for? maybe rename it to `min_energy` or something?

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/usaco-491.mdx (hunk 15 above)
Comment (Sanjana-Reddy-Kakulavaram) [reply solutions/usaco-491.mdx]: I have made all the requested changes

File: solutions/usaco-491.mdx (hunk 12 above)
Comment (SansPapyrus683) [reply solutions/usaco-491.mdx:24]: not really addressed
use better variable names instead of just slapping on a comment

Comment (envyaims) [reply solutions/usaco-491.mdx:24]: bump

**Pull request closed without merging**